spec/integration/dry/configurable/setting_spec.rb

```diff
@@ -185,6 +185,15 @@
 
     include_context "configurable behavior"
 
+    specify "settings defined after accessing config are still available in the config" do
+      klass.setting :before, default: "defined before"
+      klass.config
+      klass.setting :after, default: "defined after"
+
+      expect(klass.config.before).to eq "defined before"
+      expect(klass.config.after).to eq "defined after"
+    end
```

Comment:
This is test case that exercises the new behavior I introduced with this change.

Reply:
Perhaps it should raise in this case?